src/app/services/post.service.ts

```diff
@@ -29,14 +28,14 @@ export class PostService {
   //       Some external observer must subscribe to trigger the call.
   //       The `async` pipe performs the `.subscribe()`/`.unsubscribe()`
 
-  posts$ = this.http.get(this.allPostsUrl)
-      .map(res => res.json())
-      .mergeMap(posts=>{
-        return Observable.of(posts)
-            .merge(Observable.forkJoin(posts
-                .map(post=>this.user.getUser(post.userId)
-                    .map(user=>({...post,  authorName:user.name })))));
-      });
+  readonly posts$: Observable<PostWithAuthor[]> = this.http.get(this.allPostsUrl)
+    .flatMap(res => res.json() as Post[])
+    .mergeMap(post => this.user
+      .getUser(post.userId)
+      .map(user => ({ ...post, authorName: user.name }))
+    )
+    .toArray()
```

Comment:
Is this really necessary ?

Reply:
`toArray` is conjugation of `flatMap` on above, this way it could utilize pure RxJS operators other than jumping between `Array<T>` and `Observable<T[]>`, yet eliminate calls to static method of Observable for packing raw data around.